Use smart pointers for members in CanvasCaptureMediaStreamTrack

rwlbuis opened the pull request:
#### b36cbce69fddb7da33823f316bd8ead5bebee970
<pre>
Use smart pointers for members in CanvasCaptureMediaStreamTrack
<a href="https://bugs.webkit.org/show_bug.cgi?id=275246">https://bugs.webkit.org/show_bug.cgi?id=275246</a>

Reviewed by Sihui Liu.

Use smart pointers for members in CanvasCaptureMediaStreamTrack.

* Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.h:

Canonical link: <a href="https://commits.webkit.org/281338@main">https://commits.webkit.org/281338@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/866e1515624300374c1dc42bbc385fcec84bff49

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59434 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38777 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11955 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63348 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9945 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61563 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46431 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10109 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48256 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6998 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61464 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36215 "Found 1 new test failure: fast/forms/ios/file-upload-panel-dismiss-when-view-removed-from-window.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51443 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29080 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32924 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8694 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8881 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54875 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8974 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65081 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3362 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8893 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/55594 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3373 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51436 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55699 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13200 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2792 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34593 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35676 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36762 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35421 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.cpp:
```diff
@@ -113,12 +113,9 @@ void CanvasCaptureMediaStreamTrack::Source::requestFrameTimerFired()
-void CanvasCaptureMediaStreamTrack::Source::canvasDestroyed(CanvasBase& canvas)
+void CanvasCaptureMediaStreamTrack::Source::canvasDestroyed(CanvasBase&)
 {
-    ASSERT_UNUSED(canvas, m_canvas == &canvas);
```
Why can't we keep this?

cdumez commented on Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.cpp:
```diff
@@ -113,12 +113,9 @@ void CanvasCaptureMediaStreamTrack::Source::requestFrameTimerFired()
-    ASSERT_UNUSED(canvas, m_canvas == &canvas);
-
     stop();
-    m_canvas = nullptr;
```
ditto.

rwlbuis committed:
Use smart pointers for members in CanvasCaptureMediaStreamTrack
https://bugs.webkit.org/show_bug.cgi?id=275246

Reviewed by Sihui Liu.

Use smart pointers for members in CanvasCaptureMediaStreamTrack.

* Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.h:

Canonical link: https://commits.webkit.org/281338@main
